### PR TITLE
ADR for authn token management

### PR DESCRIPTION
**Pull request by teozkr**
Rendered: https://github.com/stackabletech/documentation/blob/adr/authn-tokens/modules/adr/pages/drafts/ADRx-authn_token_management.adoc

This is a generalization of https://github.com/stackabletech/issues/issues/4, taking into account other secret types, as well as organizationally managed secrets.

Personally I'm leaning towards doing a MVP of the custom secret service. IMO, using Secrets directly won't work well for services that care about node identities (anything that's exposed outside of the cluster), and my feeling is that relying on separate operators for each kind of secret will end up with a very disjointed experience. Vault is way too ops-heavy to have a hard dependency on, and as far as I can tell their policy framework isn't powerful enough to express our needs either.

### CODE REVIEW

**Comment by lfrancke** [comment]
Thank you for preparing this. It is a great summary. We'll talk about it in our weekly architecture meeting.

File: modules/adr/pages/ADR012-authn_token_management.adoc [hunk 1]
```diff
@@ -0,0 +1,78 @@
+= ADR012: Authentication token management
+Teo Klestrup Röijezon <teo.roijezon@stackable.de>
+v0.1, 2021-11-23
+:status: accepted
+
+* Status: {status}
+* Deciders:
```
Comment (teozkr) [inline modules/adr/pages/ADR012-authn_token_management.adoc:7]: Should this include everyone who was present in the meeting, or only the people who were actively involved in the decision?

Comment (siegfriedweber) [reply modules/adr/pages/ADR012-authn_token_management.adoc:7]: In my opinion, the actual decision makers are sufficient.

**Review by siegfriedweber — APPROVED** [review]
LGTM

File: modules/adr/pages/ADR012-authn_token_management.adoc (hunk 1 above)
Comment (lfrancke) [reply modules/adr/pages/ADR012-authn_token_management.adoc:7]: We copied the template from somewhere...I'm not super vested in any of the metadata or form :)

**Pull request merged**